diff -Naur discover-5.5.0/libdiscover/CMakeLists.txt discover-5.5.0.tpg/libdiscover/CMakeLists.txt
--- discover-5.5.0/libdiscover/CMakeLists.txt	2015-12-03 12:08:50.000000000 +0000
+++ discover-5.5.0.tpg/libdiscover/CMakeLists.txt	2015-12-12 15:19:09.864627270 +0000
@@ -38,6 +38,7 @@
 
 kconfig_add_kcfg_files(discovercommon_SRCS GENERATE_MOC MuonDataSources.kcfgc)
 add_library(DiscoverCommon ${discovercommon_SRCS})
+set_target_properties(DiscoverCommon PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${PROJECT_VERSION_MAJOR})
 
 target_link_libraries(DiscoverCommon
 LINK_PUBLIC
diff -Naur discover-5.5.0/libdiscover/notifiers/CMakeLists.txt discover-5.5.0.tpg/libdiscover/notifiers/CMakeLists.txt
--- discover-5.5.0/libdiscover/notifiers/CMakeLists.txt	2015-12-03 12:08:51.000000000 +0000
+++ discover-5.5.0.tpg/libdiscover/notifiers/CMakeLists.txt	2015-12-12 15:20:03.138100145 +0000
@@ -1,4 +1,5 @@
 add_library(DiscoverNotifiers BackendNotifierModule.cpp)
+set_target_properties(DiscoverNotifiers PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${PROJECT_VERSION_MAJOR})
 target_link_libraries(DiscoverNotifiers
     PUBLIC
         Qt5::Core
