diff -up discover-5.90.0/libdiscover/appstream/AppStreamIntegration.cpp.2~ discover-5.90.0/libdiscover/appstream/AppStreamIntegration.cpp
--- discover-5.90.0/libdiscover/appstream/AppStreamIntegration.cpp.2~	2023-11-29 11:40:45.000000000 +0100
+++ discover-5.90.0/libdiscover/appstream/AppStreamIntegration.cpp	2023-11-30 02:21:26.822424038 +0100
@@ -39,7 +39,11 @@ std::optional<AppStream::Release> AppStr
     KConfigGroup settings(KSharedConfig::openConfig(QStringLiteral("discoverrc")), u"DistroUpgrade"_s);
     bool allowPreRelease = settings.readEntry<bool>("AllowPreRelease", false);
 
-    QString currentVersion = osRelease()->versionId();
+    const QString currentVersion = osRelease()->versionId();
+    const QString currentVersionString = osRelease()->version();
+
+    if(currentVersion == QStringLiteral("cooker") || currentVersion == QStringLiteral("rolling") || currentVersion == QStringLiteral("rock") || currentVersion.endsWith(QStringLiteral(".90")) || currentVersionString.contains(QStringLiteral("Cooker")) || currentVersionString.contains(QStringLiteral("Rolling")) || currentVersionString.contains(QStringLiteral("ROME")) || currentVersionString.contains(QStringLiteral("Rome"))) // Those are current versions no matter what...
+        return std::nullopt;
     std::optional<AppStream::Release> nextRelease;
     for (const AppStream::Component &dc : distroComponents) {
 #if ASQ_CHECK_VERSION(1, 0, 0)
@@ -56,6 +60,9 @@ std::optional<AppStream::Release> AppStr
 
             // Let's look at this potentially new verson
             const QString newVersion = r.version();
+	    if (newVersion == QStringLiteral("cooker") || newVersion == QStringLiteral("rolling") || newVersion == QStringLiteral("rock")) // Don't list a different branch as a different version...
+                continue;
+
             if (AppStream::Utils::vercmpSimple(newVersion, currentVersion) > 0) {
                 if (!nextRelease) {
                     // No other newer version found yet so let's pick this one
