diff -up discover-6.3.0/libdiscover/appstream/AppStreamIntegration.cpp.2~ discover-6.3.0/libdiscover/appstream/AppStreamIntegration.cpp
--- discover-6.3.0/libdiscover/appstream/AppStreamIntegration.cpp.2~	2025-02-11 21:34:41.498135705 +0100
+++ discover-6.3.0/libdiscover/appstream/AppStreamIntegration.cpp	2025-02-11 21:44:53.838026929 +0100
@@ -57,7 +57,11 @@ std::optional<AppStream::Release> AppStr
     bool allowPreRelease = settings.readEntry<bool>("AllowPreRelease", false);
     bool allowDevelopmentRelease = settings.readEntry<bool>("AllowDevelopmentRelease", false);
 
-    QString currentVersion = osRelease()->versionId();
+    const QString currentVersion = osRelease()->versionId();
+    const QString currentVersionString = osRelease()->version();
+
+    if(currentVersion == QStringLiteral("cooker") || currentVersion == QStringLiteral("rolling") || currentVersion == QStringLiteral("rock") || currentVersion.endsWith(QStringLiteral(".90")) || currentVersionString.contains(QStringLiteral("Cooker")) || currentVersionString.contains(QStringLiteral("Rolling")) || currentVersionString.contains(QStringLiteral("ROME")) || currentVersionString.contains(QStringLiteral("Rome"))) // Those are current versions no matter what...
+        return std::nullopt;
 
     // skip if current version is unknown
     if (currentVersion.isEmpty()) {
@@ -94,6 +98,8 @@ std::optional<AppStream::Release> AppStr
 
             // Look at the potentially new version
             const QString newVersion = r.version();
+            if (newVersion == QStringLiteral("cooker") || newVersion == QStringLiteral("rolling") || newVersion == QStringLiteral("rock")) // Don't list a different branch as a different version...
+                continue;
             if (AppStream::Utils::vercmpSimple(newVersion, currentVersion) > 0) {
                 if (!nextRelease) {
                     // No other newer version found yet so let's pick this one
